manifests/init.pp

```diff
@@ -0,0 +1,188 @@
+# @summary Manage etcd
+#
+# @example
+#   include etcd
+#
+# @param version
+#   Version of etcd to install
+#   Not used if download_url is defined.
+# @param base_url
+#   Base URL of where to download etcd binaries.
+#   Not used if download_url is defined.
+# @param os
+#   The GOOS to install
+#   Not used if download_url is defined.
+# @param arch
+#   The GOARCH to install
+#   Not used if download_url is defined.
+# @param download_url
+#   Alternative location to download etcd binaries
+# @param download_dir
+#   The directory of where to download etcd
+# @param extract_dir
+#   The directory where to extract etcd
+# @param bin_dir
+#   The path to bin directory for etcd and etcdctl symlinks
+# @param manage_user
+#   Boolean that determines if etcd user is managed
+# @param manage_group
+#   Boolean that determines if etcd group is managed
+# @param user
+#   The etcd user
+# @param user_uid
+#   The etcd user UID
+# @param group
+#   The etcd group
+# @param group_gid
+#   The etcd group GID
+# @param config_path
+#   The path to etcd YAML configuration
+# @param config
+#   The config values to pass to etcd
+# @param max_open_files
+#   The value for systemd LimitNOFILE unit option
+class etcd (
+  String $version = '3.4.7',
```

Comment:
is there a way to support something like `'latest'` ? static data for the version in the module doesn't scale well.

Reply:
Nope. It's a git tag release, sort of like what Prometheus module from voxpupli does, have to specify a version.  Also the files we download are tar.gz of compiled binaries uploaded to the tag release page, so I don't think github provides a means to download that since it's not the actual code you are downloading but binaries produced after the tag.

The benefit to needing a version is if the logic needs to change and be version specific, we know we will have a version to work with for comparing `>=` or `<=` when the time comes. This is another common pattern with things like prometheus module that works very nicely.

Reply:
Too bad they don't package their software properly. I guess we're here until they have a repo.